_dev/coding-style.md

```diff
@@ -6,8 +6,65 @@ description: The preferred style for code committed to NeoMutt
 
 # Coding Style
 
-Every project has its own coding style, this one is no exception.
-However, I haven't written down the rules.  For now, ...
+## at the moment (pre-fork)
 
-- Trailing whitespace is **evil**
+Generally speaking, we don't want to diverge much from upstream mutt. We're
+still merging patches in - and if we didn't follow upstream's coding style,
+merging would be much harder.
 
+That means:
+
+* put your the braces of functions in new lines, like so:
+
+```c
+struct q_class_t
+{
+  int length;
+  int index;
+  int color;
+  char *prefix;
+  struct q_class_t *next, *prev;
+  struct q_class_t *down, *up;
+};
+```
+
+* define functions with underscores like `void mutt_account_unsetpass (..)`
+
+* what a function does *should* be clear without the need to look into ites
+  implementation - that means use expressive names!
+
+* No global variable is a good variable - therefore avoid them.
+
+* Explain with comments what a function does. That includes:
+
+    - the return value and their meaning
+
+    - parameters and their meaning
+
+
+* Use only `/* Comment */` style comments. Don't use single line comments (`//
+  Comment`)
+
+* use a two spaces based indentation
+
+* Wrap your code and comments at 80 Columns.
```

Comment:
There again, we should be relaxed, if a line of code makes 82 columns, I believe wrapping it would be abusive. Even upstream mutt is relaxed on this.

The rationale is to avoid all code being formatted at 150 columns, or have super long one liners.